### PR TITLE
feat:: V4 tokens tab, "Reserved tokens" subpanel [4/n]

### PR DESCRIPTION
**Pull request by johnnyd-eth**
Screenshot 2024-07-19 at 9 30 42 pm

### CODE REVIEW

File: src/packages/v2v3/components/shared/SplitList.tsx [hunk 1]
```diff
@@ -4,6 +4,19 @@ import { getProjectOwnerRemainderSplit, sortSplits } from 'packages/v2v3/utils/v
 import { useMemo } from 'react'
 import { SplitItem, SplitProps } from './SplitItem'
 
+export type V2V3SplitListProps = {
```
Comment (aeolianeth) [inline src/packages/v2v3/components/shared/SplitList.tsx:7]: 🙏

File: src/packages/v4/components/V4SplitList.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { BigNumber } from "@ethersproject/bignumber"
+import SplitList, { V2V3SplitListProps } from "packages/v2v3/components/shared/SplitList"
+import { V4Split } from "../models/v4Split"
+import { v4SplitToV2V3Split } from "../utils/v4Splits"
+
+type V4SplitListProps = Omit<V2V3SplitListProps, 'splits' | 'currency' | 'totalValue'> & {
```
Comment (aeolianeth) [inline src/packages/v4/components/V4SplitList.tsx:6]: Hmm. I think we should keep v4 totally discinct from v3. WDYT about just recreating the type script scratch here?

File: src/packages/v4/components/V4SplitList.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { BigNumber } from "@ethersproject/bignumber"
+import SplitList, { V2V3SplitListProps } from "packages/v2v3/components/shared/SplitList"
```
Comment (aeolianeth) [inline src/packages/v4/components/V4SplitList.tsx:2]: IMO we should either figure out how to make this truly generic component (not v2v3-related), or just duplicate it and modify for v4. I think we should keep v4 and v2v3 as separated as possible.

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/V4DistributeReservedTokensModal.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
+import { t, Trans } from '@lingui/macro'
+import TransactionModal from 'components/modals/TransactionModal'
+import useNameOfERC20 from 'hooks/ERC20/useNameOfERC20'
+import { useJBContractContext, useReadJbTokensTokenOf, useReadJbTokensTotalCreditSupplyOf } from 'juice-sdk-react'
+import { useDistributeReservedTokens } from 'packages/v2v3/hooks/transactor/useDistributeReservedTokensTx'
```
Comment (aeolianeth) [inline src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/V4DistributeReservedTokensModal.tsx:5]: Needs a nuke i think?

Honestly at this stage, i would completely not worry about any transaction-related features. I can take them 🙏 , need to establish a Wagmi way of doing txs first (and that work with the modals)

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/hooks/useV4ReservedTokensSubPanel.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import { WeiPerEther } from '@ethersproject/constants'
+import { ONE_BILLION } from 'constants/numbers'
+import { useJBContractContext, useJBRulesetMetadata, useReadJbTokensTotalCreditSupplyOf } from 'juice-sdk-react'
+import useProjectOwnerOf from 'packages/v4/hooks/useV4ProjectOwnerOf'
+import { useV4ReservedSplits } from 'packages/v4/hooks/useV4ReservedSplits'
+import { formatV4SplitPercent } from 'packages/v4/utils/math'
```
Comment (aeolianeth) [inline src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/hooks/useV4ReservedTokensSubPanel.ts:6]: Can we pls remove all uses of this and replace it with `SplitPortion` from the SDK?

`new SplitPortion(value).format()`

**Review by aeolianeth — APPROVED** [review]

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/hooks/useV4ReservedTokensSubPanel.ts (hunk 5 above)
Comment (johnnyd-eth) [reply src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/hooks/useV4ReservedTokensSubPanel.ts:6]: Done

File: src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/V4DistributeReservedTokensModal.tsx (hunk 4 above)
Comment (johnnyd-eth) [reply src/packages/v4/views/V4ProjectDashboard/V4ProjectTabs/V4TokensPanel/V4DistributeReservedTokensModal.tsx:5]: Got it 👍

File: src/packages/v4/components/V4SplitList.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/packages/v4/components/V4SplitList.tsx:6]: On second look it's definitely a good call. I copied all the SplitsList and SplitItem components over to the V4 package and adjusted for V4 👍

File: src/packages/v4/components/V4SplitList.tsx (hunk 3 above)
Comment (johnnyd-eth) [reply src/packages/v4/components/V4SplitList.tsx:2]: Addressed https://github.com/jbx-protocol/juice-interface/pull/4399#discussion_r1685956311

**Pull request merged**